common_core.c

```diff
@@ -53,7 +54,7 @@ int ta_generate_address(const iota_client_service_t* const service,
   if (ret == RC_OK) {
     res->addresses = out_address;
   }
-  hash243_queue_free(&out_address);
+  out_address = NULL;
```

Comment:
Since `out_address` is just a local variable, it is meaningless to set its value by the end of function `ta_generate_address`.